File: DSA/ManasMaheshwari_590015617/Day_4/Question1_LeetCode.java
// LeetCode Problem 867: Transpose Matrix

class Solution {
    public int[][] transpose(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        
        int[][] new_matrix = new int[n][m];
        
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                new_matrix[j][i] = matrix[i][j];
            }
        }
        
        return new_matrix;
    }
}

File: DSA/Divyanshi_590017786/Day_4/Question1_Leetcode.py
class Solution:
    def transpose(self, matrix):
        m = len(matrix)
        n = len(matrix[0])

        ans = [[0] * m for _ in range(n)]

        for i in range(m):
            for j in range(n):
                ans[j][i] = matrix[i][j]

        return ans